soroban-env-host/src/native_contract/token/event.rs

```diff
@@ -0,0 +1,33 @@
+use crate::host::Host;
+use crate::native_contract::base_types::{BigInt, Vec};
+use crate::native_contract::token::error::Error;
+use crate::native_contract::token::public_types::Identifier;
+use soroban_env_common::{CheckedEnv, Symbol, TryIntoVal};
+
+pub(crate) fn approve(
+    e: &Host,
+    from: Identifier,
+    to: Identifier,
+    amount: BigInt,
+) -> Result<(), Error> {
+    let mut topics = Vec::new(e)?;
+    topics.push(Symbol::from_str("approve"))?;
+    topics.push(from)?;
+    topics.push(to)?;
+    e.contract_event(topics.into(), amount.try_into_val(e)?)?;
+    Ok(())
+}
+
+pub(crate) fn transfer(
+    e: &Host,
+    from: Identifier,
+    to: Identifier,
+    amount: BigInt,
+) -> Result<(), Error> {
+    let mut topics = Vec::new(e)?;
+    topics.push(Symbol::from_str("xfer"))?;
```

Comment:
Should this be `transfer` instead? It's also used in `xfer_from`, so `xfer` could be confusing.

Reply:
Yeah good point. I will fix.

Reply:
Fixed.